bin/add_frontend_server.rb

```diff
@@ -98,3 +165,43 @@
 private_dns_name = ec2client.describe_instances({instance_ids: [instance_id],}).reservations[0].instances[0].private_dns_name
 
 puts "Private DNS name #{private_dns_name}"
+
+puts 'Writing new configuration file'
+
+Net::SSH.start('gateway.code.org', username, :password => password) do |ssh|
+  ssh.exec!("knife environment show #{options['environment']} -F json > /tmp/old_knife_config")
+end
+
+Net::SCP.download!('gateway.code.org', username, '/tmp/old_knife_config', '/tmp/knife_config', :ssh => { :password => password })
+
+configuration_json = JSON.parse(File.read('/tmp/knife_config'))
+configuration_json['override_attributes']['cdo-secrets']['app_servers'] ||= {}
+configuration_json['override_attributes']['cdo-secrets']['app_servers'][instance_name] = private_dns_name
+
+File.open('/tmp/new_knife_config.json', 'w') do |f|
+  f.write(JSON.dump(configuration_json))
+end
+
+Net::SCP.upload!('gateway.code.org', username, '/tmp/new_knife_config.json', '/tmp/new_knife_config.json', :ssh => { :password => password })
+puts 'New configuration file written.'
+
+Net::SSH.start('gateway.code.org', username, :password => password) do |ssh|
+
+  ssh.open_channel do |ch|
+    execute_ssh_on_channel(ch,
+                           "knife environment from file /tmp/new_knife_config.json",
```

Comment:
I'd like to be able to also simplify this block as well to avoid the filesystem/SCP, but it seems that Chef somehow doesn't support reading configuration from standard input. (Specifically, `knife` requires its `from file` command to have a `.json` file extension, which makes doing `knife environment from file /dev/stdin` fail. `ln -s -f /dev/stdin /tmp/stdin.json && knife environment from file /tmp/stdin.json` works as a workaround, but is quite a hack and I'd rather not resort to this here.) Probably can leave it as-is for now (I'll file an issue in Chef to fix/support this workflow in the future).